core/domain/question_services.py

```diff
@@ -201,3 +233,73 @@ def update_question(committer_id, question_id, change_list, commit_message):
     """
     question = apply_change_list(question_id, change_list)
     _save_question(committer_id, question, change_list, commit_message)
+
+
+def add_question_id_to_skill(question_id, collection_id, skill_id, user_id):
+    """Adds the question id to the question list of the appropriate skill.
+
+    Args:
+        question_id: str. The id of the question.
+        collection_id: str. The id of the collection.
+        skill_id: str. The id of the skill.
+        user_id: str. The id of the user.
+    """
+    collection_services.update_collection(
+        user_id, collection_id, [{
+            'cmd': collection_domain.CMD_ADD_QUESTION_ID_TO_SKILL,
+            'skill_id': skill_id,
+            'question_id': question_id
+        }], 'Add a new question with ID %s to skill with ID %s'%(
+            question_id, skill_id))
+
+
+def remove_skill(question_id, collection_id, skill_id, user_id):
+    """Removes the question id from the question list of the appropriate
+    skill.
+
+    Args:
+        skill_id: str. The id of the skill.
+        user_id: str. The id of the user.
+    """
+    collection_services.update_collection(
+        user_id, collection_id, [{
+            'cmd': collection_domain.CMD_REMOVE_QUESTION_ID_FROM_SKILL,
+            'skill_id': skill_id,
+            'question_id': question_id
+        }], 'Remove a question with ID %s from skill with ID: %s'%(
+            question_id, skill_id))
+
+
+def get_questions_batch(
+        collection_id, skill_ids, user_id, batch_size):
+    """Fetches a batch of questions for a user based on the provided
+    skill_ids, after filtering to only include skills that the user
+    has already acquired.
+
+    Args:
+        collection_id: str. Id of the collection.
+        skill_ids: list(str). A list of skill IDs for the questions to
+            be retrieved.
+        user_id: str. Id of the user.
+        batch_size: int. The intended number of questions to be returned.
+
+    Returns:
+        list(dict). A list of Question object dicts.
```

Comment:
Better to return a list of question objects.

Reply:
Updated.